### PR TITLE
add aggregated statistics diagram

### PR DESCRIPTION
**Pull request by thrillfall**
**Describe the pull request**
Instant post-workout track export to external storage

**Link to the the issue**
#186 #90

### CODE REVIEW

**Comment by dennisguse** [comment]
Nice work! 
Just one (not so tiny thing) - could you reuse the ChartView that we already have?
So far, we are (almost) dependency free.

**Comment by thrillfall** [comment]
i tried using ChartView but that is just not what i can handle. 
Correct me if i am wrong but ChartView works with a List of ChartPoint. These ChartPoints are a representation of tracked locations which is not something i can provide.
My chart works with a set of 2d vectors computated from several tracks. Trying to force this computated data into a structure that does something totally different (render a set of locations) seems totally wrong and a lot of effort.

TL;DR;  i don't think its a good idea to use ChartView for the graph i want to provide.

**Comment by rgmf** [comment]
Great idea!!! And huge job!!! :D

> Currently this is an additional menu entry in settings. I could imagine converting the existing aggregated stats activity and this to a tabbed view. what do you think?

IMHO the best option would be added "fastest average speed" and "greatest distance" in the `ListView` that is in `AggregatedStatisticsActivity` just below of "Avg. moving pace" and "Fastest pace" for every type of activity.

Also, I'd add the `onClick` listener to each item of that `ListView` so user clicks on "TRAIL RUNNING" section then `ShowWorkoutsAggregatedDiagramActivity` is opened with diagrams. In this way user doesn't need select type of activity (in the spinner) because s-he selects it clicking on the item s-he want to see.

What do you think @thrillfall?

Finally, I've checked `ShowWorkoutsAggregatedDiagramActivity` out with my actual tracks and I see this:

Captura de pantalla_2020-08-16_16-28-13

That it's very difficult to analize.

I think, this diagrams should showed by week because we have a space limit for show all activities.

The long story short (IMHO):
- Diagrams by week: focused on activities.
- Diagrams by month: focused on aggregated stats by week (4 or 5 weeks).
- Diagrams by year: focused on aggregated stats by month (12 months).

I know it's a huge job but I think the result will be great.

I comment all this because I was thinking about all that for several weeks ago, but it's only an opinion and it can be focused from a lot of perspectives.

What do you think? @thrillfall @dennisguse

**Comment by dennisguse** [comment]
About the general data presentation: I don't really have an opinion about that as I rarely need it.
But for our users: it would be awesome!
And we even have some requests to improve our data representation (like #185).

And I have another proposal (for discussion): #379.

**Comment by thrillfall** [comment]
> That it's very difficult to analize.
@rgmf You are right. But this weird curving speed graph is a bug. Somehow the values for the date do not align for speed and distance point on the graph.

Thank you also foir all the nice ideas.
I'd also favor moving the maximum distance and speed (with date) to the existing view.

For now users would (given the bug is fixed) have the option to zoom in/out and go back and forth in the graph to see the results for the time range in view. 
As for your idea to have different aggregations for different time ranges i'd see that as a topic for another increment and not as a show stopper for this MR. Lets give the other users the graph as is (without that bug) and extend it from release to release.

**Comment by thrillfall** [comment]
> i tried using ChartView but that is just not what i can handle.
> Correct me if i am wrong but ChartView works with a List of ChartPoint. These ChartPoints are a representation of tracked locations which is not something i can provide.
> My chart works with a set of 2d vectors computated from several tracks. Trying to force this computated data into a structure that does something totally different (render a set of locations) seems totally wrong and a lot of effort.
> 
> TL;DR; i don't think its a good idea to use ChartView for the graph i want to provide.

so can i assume from your response that i'd be an option to use the given approach for the graph? I would neither have the time nor the will to force this into ChartView.

**Comment by dennisguse** [comment]
I would love this - only thing is the technical implementation.
And if we find a way to kick/replace the current charts: I am all in.
My general concern is maintainability (@thrillfall that's my issue with external libs).

@thrillfall just go ahead - we will figure something out.
And by the way: thanks for your time that you already spent!

**Comment by thrillfall** [comment]
@rgmf the weird curve was not so much a bug but a special mode i choose for displaying the speed graph. I changed that back to the default linear mode which should make the graph more readable
Screenshot_20200816-230127_OpenTracks_(Debug)

**Comment by rgmf** [comment]
Great @thrillfall ;)

I'll test it tomorrow to see the results with all my tracks but seem more much better :D
